src/ESLint.Formatter/sarif-with-rules.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * @fileoverview SARIF v2.0 formatter
+ * @author Chris Meyer
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Helper Functions
+//------------------------------------------------------------------------------
+
+/**
+ * Returns the severity of warning or error
+ * @param {Object} message message object to examine
+ * @returns {string} severity level
+ * @private
+ */
+function getResultLevel(message) {
```

Comment:
>getResultLevel [](start = 9, length = 14)

isn't this code duplicated with another helper in sarif.js? can we do some code sharing?

Reply:
This is a version of the formatter that is ready for the design change I have proposed. The primary purpose of this PR is to get this work into source control.

---
In reply to: [249138474](https://github.com/Microsoft/sarif-sdk/pull/1222#discussion_r249138474) [](ancestors = 249138474)